pkg/rbd/controllerserver.go

```diff
@@ -698,6 +703,7 @@ func (cs *ControllerServer) ControllerExpandVolume(ctx context.Context, req *csi
 	defer cr.DeleteCredentials()
 
 	rbdVol := &rbdVolume{}
+	defer rbdVol.Destroy()
```

Comment:
instead of calling Destroy on Each function, we can call in in one function (maybe `DeleteCredentials` function) this   need to be taken care in next patch

Reply:
This can easily be done after https://github.com/nixpanic/ceph-csi/commit/c304e9220b0665c234037a37ed94e7381db66158 as then the credentials are only passed to `rbVol.Connect(cr)`. `createImage()` can be refactored quite a bit then too.